### PR TITLE
Android Java code now uses webcompat_report_api_endpoint arg

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
This PR brings `WEBCOMPAT_REPORT_ENDPOINT` from `.npmrc` into `BraveConfig.java` so Android uses now the same url as desktop.

Also was fixed failed assertion at `Profile.getLastUsedRegularProfile()` from non-UI thread, so it was impossible to send feedback from debug build.

During my tests I saw SUCCESS status just once and many times I saw status 500 (internal server error)  - with both new (`https://webcompat.brave.com/1/webcompat`) and old (`https://laptop-updates.brave.com/1/webcompat`) urls. Maybe it happened because `brave_stats_api_key` is required, but I don't have it.

Besides all this PR fixes build-time warning introduced at https://github.com/brave/brave-core/pull/18129
```
WARNING at the command-line "--args":1:2314: Build argument has no effect.
webcompat_report_api_endpoint="https://webcompat.brave.com/1/webcompat"
                              ^----------------------------------------
The variable "webcompat_report_api_endpoint" was set as a build argument
but never appeared in a declare_args() block in any buildfile.
```

Resolves https://github.com/brave/brave-browser/issues/30034

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Open some website
2. Turn the shields off
3. Press `Report a broken site`, expected not to crash and see `Thank you` message.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java [hunk 1]
```diff
@@ -831,7 +832,12 @@ public void onClick(View view) {
         mSubmitButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                BraveShieldsUtils.BraveShieldsWorkerTask mWorkerTask = new BraveShieldsUtils.BraveShieldsWorkerTask(mTitle);
+                // Profile.getLastUsedRegularProfile requires to run in UI thread,
+                // so get aip key here and pass it to IO worker task
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java:836]: nit: api key

**Review by samartnik — APPROVED** [review]
lgtm

**Review by porteron — APPROVED** [review]

**Review by antonok-edm — APPROVED** [review]
awesome, thanks!

**Review by deeppandya — APPROVED** [review]
LGTM. Thank you @AlexeyBarabash

**Pull request merged**